File: 프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.c
#include <stdio.h>
#include <stdbool.h>
#include <stdlib.h>

int solution(int num) {
    int i = 0;
    long long n = num;
    for(i = 0;i <= 500;i++){
        if(n == 1){ break;}
        if(n % 2 == 0){n /= 2;}
        else{
            n = 3*n + 1;
        }
    }
    if(i > 500){return -1;}
    return i;
}

File: 프로그래머스/1/12943.콜라츠 추측-Python3/콜라츠 추측.py
def solution(num):
    answer = 0
    for i in range(500):
        if num == 1:
            break

        answer += 1

        if num % 2 == 0:
            num //= 2
        else:
            num = 3*num + 1

    if answer == 500:
        return -1

    return answer